### PR TITLE
Feature/RM-9029 optimize wxe context factory and wxe context mock

### PR DESCRIPTION
**Pull request by Frazerus**
https://re-motion.atlassian.net/browse/RM-9029

### CODE REVIEW

**Review by MichaelKetting — COMMENTED** [review]
Waiting with review until RM-8989 is merged and this is rebased on develop to reduce noise in changes-list

File: Remotion/Development/Web/Development.Web.csproj [hunk 1]
```diff
@@ -19,5 +19,8 @@
     <ProjectReference Include="..\..\Web\Core\Web.Core.csproj" />
     <ProjectReference Include="..\Core\Development.Core.csproj" />
   </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="Moq" Version="4.16.1" />
+  </ItemGroup>
```
Comment (MichaelKetting) [inline Remotion/Development/Web/Development.Web.csproj:24]: This dependency is not allowed in a library.

**Review by MichaelKetting — CHANGES_REQUESTED** [review]

**Comment by Frazerus** [comment]
Ah sorry, i thought github was buggy, didnt realize you already reviewed again

File: Remotion/Web/UnitTests/Core/ExecutionEngine/WxeFunctionTests/WxeFunctionTest.cs [hunk 2]
```diff
@@ -76,8 +76,7 @@ public void SetTransactionMode ()
       function.Add(new WxeDelegateStep(() => actualTransaction = function.Transaction));
       function.SetTransactionMode(WxeTransactionMode<TestTransactionFactory>.CreateRoot);
 
-      WxeContextFactory contextFactory = new WxeContextFactory();
-      var context = contextFactory.CreateContext(function);
+      var context = WxeContextFactory.Create(new TestFunction());
```
Comment (MichaelKetting) [inline Remotion/Web/UnitTests/Core/ExecutionEngine/WxeFunctionTests/WxeFunctionTest.cs:79]:
```suggestion
      var context = WxeContextFactory.Create(function);
```

File: Remotion/Web/UnitTests/Core/ExecutionEngine/WxeFunctionTests/WxeFunctionTest.cs [hunk 3]
```diff
@@ -97,8 +96,7 @@ public void SetTransactionMode_AfterExecutionHasStarted_ThrowsInvalidOperationEx
                   Throws.InvalidOperationException
                       .With.Message.EqualTo("The TransactionMode cannot be set after the TransactionStrategy has been initialized."))));
 
-      WxeContextFactory contextFactory = new WxeContextFactory();
-      var context = contextFactory.CreateContext(function);
+      var context = WxeContextFactory.Create(new TestFunction());
```
Comment (MichaelKetting) [inline Remotion/Web/UnitTests/Core/ExecutionEngine/WxeFunctionTests/WxeFunctionTest.cs:99]:
```suggestion
      var context = WxeContextFactory.Create(function);
```

File: Remotion/Web/UnitTests/Core/ExecutionEngine/WxeFunctionTests/WxeFunctionTest.cs [hunk 4]
```diff
@@ -136,8 +134,7 @@ public void SetExecutionListener_AfterExecutionHasStarted_ThrowsInvalidOperation
                   Throws.InvalidOperationException
                       .With.Message.EqualTo("The ExecutionListener cannot be set after the TransactionStrategy has been initialized."))));
 
-      WxeContextFactory contextFactory = new WxeContextFactory();
-      var context = contextFactory.CreateContext(function);
+      var context = WxeContextFactory.Create(new TestFunction());
```
Comment (MichaelKetting) [inline Remotion/Web/UnitTests/Core/ExecutionEngine/WxeFunctionTests/WxeFunctionTest.cs:137]:
```suggestion
      var context = WxeContextFactory.Create(function);
```

File: Remotion/Web/UnitTests/Core/ExecutionEngine/WxeFunctionTests/WxeFunctionTest.cs [hunk 5]
```diff
@@ -165,8 +162,7 @@ public void GetExecutingStep_AfterExecutionHasFinished_ReturnsCurrentFunction ()
       TestFunction2 function = new TestFunction2();
       function.Add(new WxeDelegateStep(() => { }));
 
-      WxeContextFactory contextFactory = new WxeContextFactory();
-      var context = contextFactory.CreateContext(function);
+      var context = WxeContextFactory.Create(new TestFunction());
```
Comment (MichaelKetting) [inline Remotion/Web/UnitTests/Core/ExecutionEngine/WxeFunctionTests/WxeFunctionTest.cs:165]:
```suggestion
      var context = WxeContextFactory.Create(function);
```

File: Remotion/Web/UnitTests/Core/ExecutionEngine/WxeFunctionTests/WxeFunctionTest.cs [hunk 6]
```diff
@@ -186,8 +182,7 @@ public void GetExecutingStep_AfterExecutionHasStarted_ReturnsCurrentStep ()
           });
       function.Add(expectedStep);
 
-      WxeContextFactory contextFactory = new WxeContextFactory();
-      var context = contextFactory.CreateContext(function);
+      var context = WxeContextFactory.Create(new TestFunction());
```
Comment (MichaelKetting) [inline Remotion/Web/UnitTests/Core/ExecutionEngine/WxeFunctionTests/WxeFunctionTest.cs:185]:
```suggestion
      var context = WxeContextFactory.Create(function);
```

File: Remotion/Web/UnitTests/Core/ExecutionEngine/WxeFunctionTests/WxeFunctionTest.cs [hunk 7]
```diff
@@ -210,8 +205,7 @@ public void GetLastExecutedStep_AfterExecutionHasFinished_ReturnsNull ()
       TestFunction2 function = new TestFunction2();
       function.Add(new WxeDelegateStep(() => { }));
 
-      WxeContextFactory contextFactory = new WxeContextFactory();
-      var context = contextFactory.CreateContext(function);
+      var context = WxeContextFactory.Create(new TestFunction());
```
Comment (MichaelKetting) [inline Remotion/Web/UnitTests/Core/ExecutionEngine/WxeFunctionTests/WxeFunctionTest.cs:208]:
```suggestion
      var context = WxeContextFactory.Create(function);
```

File: Remotion/Web/UnitTests/Core/ExecutionEngine/WxeFunctionTests/WxeFunctionTest.cs [hunk 8]
```diff
@@ -231,8 +225,7 @@ public void GetLastExecutedStep_AfterExecutionHasStarted_ReturnsNotNull ()
           });
       function.Add(expectedStep);
 
-      WxeContextFactory contextFactory = new WxeContextFactory();
-      var context = contextFactory.CreateContext(function);
+      var context = WxeContextFactory.Create(new TestFunction());
```
Comment (MichaelKetting) [inline Remotion/Web/UnitTests/Core/ExecutionEngine/WxeFunctionTests/WxeFunctionTest.cs:228]:
```suggestion
      var context = WxeContextFactory.Create(function);
```

File: Remotion/Development/Web/UnitTesting/ExecutionEngine/WxeContextFactory.cs [hunk 9]
```diff
@@ -44,26 +44,30 @@ public static HttpContext CreateHttpContext ()
...
-        : base(
-            new HttpContextWrapper(context),
-            new WxeFunctionStateManager(new HttpSessionStateWrapper(context.Session)),
-            new WxeFunctionState(new TestFunction(), 20, false),
-            null!,
-            wxeUrlSettings,
-            wxeLifetimeManagementSettings)
+    public static WxeContext Create (HttpContext context, WxeUrlSettings wxeUrlSettings, IWxeLifetimeManagementSettings wxeLifetimeManagementSettings)
     {
+      return Create(context, null!, wxeUrlSettings, wxeLifetimeManagementSettings);
```
Comment (MichaelKetting) [inline Remotion/Development/Web/UnitTesting/ExecutionEngine/WxeContextFactory.cs:49]:
```suggestion
      return new WxeContext(
          new HttpContextWrapper(context),
          new WxeFunctionStateManager(new HttpSessionStateWrapper(context.Session)),
          new WxeFunctionState(new TestFunction(), 20, false),
          null!,
          wxeUrlSettings,
          wxeLifetimeManagementSettings);
```

**Review by MichaelKetting — APPROVED** [review]
Small changes, all in comments.

@a-ctor could you please push an update for this PR since @Frazerus isn't available right now and we can close it.

**Pull request merged**